Description: Fix UnboundLocalError for logger.
 Attempt to avoid unbound local and unbounded recursion errors when
 initialising the logger.
 .
 moin (2.0+git20220927.beec62a5-1) unstable; urgency=medium
 .
   * Initial release.
Author: Paul Boddie <paul@boddie.org.uk>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2022-07-03

--- moin-2.0.orig/src/moin/log.py
+++ moin-2.0/src/moin/log.py
@@ -120,6 +120,9 @@ def load_config(conf_fname=None):
     """ load logging config from conffile """
     global configured
     err_msg = None
+    was_configured = configured
+    configured = True
+    logger = getLogger(__name__)
     conf_fname = os.environ.get('MOINLOGGINGCONF', conf_fname)
     if conf_fname:
         try:
@@ -133,18 +136,14 @@ def load_config(conf_fname=None):
                 logging.config.fileConfig(f)
             finally:
                 f.close()
-            configured = True
-            logger = getLogger(__name__)
             logger.debug('using logging configuration read from "{0}"'.format(conf_fname))
             warnings.showwarning = _log_warning
         except Exception as err:  # XXX be more precise
             err_msg = str(err)
-    if not configured:
+    if not was_configured:
         # load builtin fallback logging config
         with StringIO(logging_config) as f:
             logging.config.fileConfig(f)
-        configured = True
-        logger = getLogger(__name__)
         if err_msg:
             logger.warning('load_config for "{0}" failed with "{1}".'.format(conf_fname, err_msg))
         logger.debug('using logging configuration read from built-in fallback in moin.log module!')
